### ISSUE TITLE
Sometimes when scrolling through table rows, I receive warning: Failed prop type: The prop `index` is marked as required in `FixedDataTableRowImpl`, but its value is `null`.

### ISSUE THREAD

**Issue by pauliusg** [schrodinger/fixed-data-table-2#644]
In my case I receive warning when there are more rows to fit in the screen and I scroll to the bottom then start to scroll back to the top. In most times this is enough to get warning but not always.

```
index.js:1 Warning: Failed prop type: The prop `index` is marked as required in `FixedDataTableRowImpl`, but its value is `null`.
    at FixedDataTableRowImpl (http://localhost:3011/merch/static/js/vendors~main.chunk.js:46437:5)
    at FixedDataTableRow (http://localhost:3011/merch/static/js/vendors~main.chunk.js:46917:5)
    at div
    at FixedDataTableBufferedRows (http://localhost:3011/merch/static/js/vendors~main.chunk.js:44209:5)
    at div
    at div
    at FixedDataTable (http://localhost:3011/merch/static/js/vendors~main.chunk.js:42937:5)
    at ScrollContainer (http://localhost:3011/merch/static/js/vendors~main.chunk.js:47930:5)
    at FixedDataTableContainer (http://localhost:3011/merch/static/js/vendors~main.chunk.js:46105:5)
    at Table (http://localhost:3011/merch/static/js/main.chunk.js:9559:3)
    at div
    at DynamicWidth (http://localhost:3011/merch/static/js/main.chunk.js:9774:3)
    at TableContainer (http://localhost:3011/merch/static/js/main.chunk.js:10142:3)
    at Ue (http://localhost:3011/merch/static/js/main.chunk.js:11531:9)
    at Ge (http://localhost:3011/merch/static/js/main.chunk.js:11605:13)
    at Route (http://localhost:3011/merch/static/js/vendors~main.chunk.js:168474:29)
    at Switch (http://localhost:3011/merch/static/js/vendors~main.chunk.js:168676:29)
    at div
    at push.../../ws-core/ws-core-ui/dist/index.js.exports.TaskRoutes (http://localhost:3011/merch/static/js/main.chunk.js:11904:13)
    at TaskRoutes (http://localhost:3011/merch/static/js/main.chunk.js:101258:3)
    at Route (http://localhost:3011/merch/static/js/vendors~main.chunk.js:168474:29)
    at Switch (http://localhost:3011/merch/static/js/vendors~main.chunk.js:168676:29)
    at ErrorBoundary (http://localhost:3011/merch/static/js/main.chunk.js:95757:5)
    at Oe (http://localhost:3011/merch/static/js/main.chunk.js:11312:13)
    at Re (http://localhost:3011/merch/static/js/main.chunk.js:11369:13)
    at push.../../ws-core/ws-core-ui/dist/index.js.exports.CoreContainer (http://localhost:3011/merch/static/js/main.chunk.js:11878:13)
    at AdminLayout (http://localhost:3011/merch/static/js/main.chunk.js:94648:1)
    at AdminRouter (http://localhost:3011/merch/static/js/main.chunk.js:95092:1)
    at Route (http://localhost:3011/merch/static/js/vendors~main.chunk.js:168474:29)
    at ProtectedRouteBase (http://localhost:3011/merch/static/js/main.chunk.js:7125:1)
    at ProtectedRoute (http://localhost:3011/merch/static/js/main.chunk.js:66514:5)
    at WithAppContextFunctions (http://localhost:3011/merch/static/js/main.chunk.js:69329:7)
    at Switch (http://localhost:3011/merch/static/js/vendors~main.chunk.js:168676:29)
    at http://localhost:3011/merch/static/js/vendors~main.chunk.js:117447:23
    at AdminRoot
    at Router (http://localhost:3011/merch/static/js/vendors~main.chunk.js:168105:30)
    at Provider (http://localhost:3011/merch/static/js/vendors~main.chunk.js:165342:20)
    at AppComponent
    at ContextProvider (http://localhost:3011/merch/static/js/vendors~main.chunk.js:54205:25)
```

If, I look at FixedDataTableBufferedRows.js code, I see that index can be null:
https://github.com/schrodinger/fixed-data-table-2/blob/9aa115c5797dc314e76963525dd4d383b0fe7696/src/FixedDataTableBufferedRows.js#L130-L138

Debugging also approves this situation:
![image](https://user-images.githubusercontent.com/1721818/175011330-47b369aa-0665-4f3f-be47-439336ab0cbd.png)
![image](https://user-images.githubusercontent.com/1721818/175011391-a6109954-8096-46d4-b9b5-4ec5af82361d.png)

The biggest issue is not a warning in FixedDataTableRow.js but that table cell with rowIndex=null is rendered:
https://github.com/schrodinger/fixed-data-table-2/blob/ad60599a46015280452c85851a7c20dac3cf606a/src/FixedDataTableCell.js#L310
![image](https://user-images.githubusercontent.com/1721818/175013996-3eacbddb-3de3-4288-9fe4-de24cf44efce.png)

**When documentation in https://github.com/schrodinger/fixed-data-table-2 says:**
The cell components in a column will receive the current array index of your data as a prop (this.props.rowIndex). Use this to access the correct value for each cell.
```
const rows = [0,1,2];
    
<Column
  header={<Cell>Column 1</Cell>}
  cell={({rowIndex, ...props}) => (
    <Cell {...props}>
    {rows[rowIndex]}
    </Cell>
  )}
  width={2000}
/>
```

So, in my case I get rows[rowIndex]=undefined...

I can fix this for my case, **but I guess that in such case cell renderer should not be called.**

**Comment by pauliusg** [comment]
Hi @pradeepnschrodinger, thank you for quick response. Unfortunately, I don't have more time to make PR. I am using latest FDT version 1.2.9.

**Comment by tkirda** [comment]
Hi @pradeepnschrodinger, you were correct. There is a flaw in logic when assigning rowIndex. In case when _staticRowArray[i] already assigned to null, we should keep it as undefined so it can skip rendering that row.

**Comment by pradeepnschrodinger** [comment]
This should now be resolved with [v1.2.10](https://github.com/schrodinger/fixed-data-table-2/releases/tag/v1.2.10) through #645.

Thanks @pauliusg and @tkirda!
